pkg/stream/consumer_base.go

```diff
@@ -58,14 +59,15 @@ type baseConsumer struct {
 	kernel.ApplicationStage
 	ConsumerAcknowledge
 
-	clock        clock.Clock
-	uuidGen      uuid.Uuid
-	logger       log.Logger
-	metricWriter metric.Writer
-	tracer       tracing.Tracer
-	encoder      MessageEncoder
-	retryInput   Input
-	retryHandler RetryHandler
+	clock          clock.Clock
+	uuidGen        uuid.Uuid
+	logger         log.Logger
+	metricWriter   metric.Writer
+	tracer         tracing.Tracer
+	encoder        MessageEncoder
+	retryInput     Input
+	retryHandler   RetryHandler
+	retryGraceTime time.Duration
```

Comment:
Is there a reason which I don't see why we have this `retryGraceTime` as a dedicated field?\
don't we have this one on the consumer settings on line 79 already?\

Reply:
Good catch. I introduced a new field because I first took it from the kernel directly, but now it is in the settings I already have. Removed it again and used it directly from the settings.